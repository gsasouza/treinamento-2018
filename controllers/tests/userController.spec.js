const request = require('supertest');
const app = require('../../app');
const User = require('../../models/userModel');
const { sanitizeTestObject } = require('../../test/sanitizeObject');

beforeAll(() => User.remove({}));

describe('Users', () => {

  describe('Create', () => {
    test('Should create a user', async () => {
      const response = await request(app).post('/user').send(
        {
          username: 'test',
          password: 'test123',
          name: 'name',
        }
      );
      expect(response.status).toBe(201);
      expect(sanitizeTestObject(response.body)).toMatchSnapshot();
    });

    test('Should not create a user with the same username', async () => {
      const response = await request(app).post('/user').send(
        {
          username: 'test',
          password: 'test123',
          name: 'name',
        }
      );
      expect(response.status).toBe(500);
      expect(sanitizeTestObject(response.body)).toMatchSnapshot();
    });
    test('Should not create a without username', async () => {
      const response = await request(app).post('/user').send(
        {
          password: 'test123',
          name: 'name',
        }
      );
      expect(response.status).toBe(500);
      expect(sanitizeTestObject(response.body)).toMatchSnapshot();
    })
  });

  describe('Find', () => {
    test('Should find a user', async () => {
      const user = await User.create(        {
        username: 'test2',
        password: 'test123',
        name: 'name',
      });
      const response = await request(app).get(`/user/${user.username}`);
      expect(response.status).toBe(200);
      expect(sanitizeTestObject(response.body)).toMatchSnapshot();
    });
  })

});
