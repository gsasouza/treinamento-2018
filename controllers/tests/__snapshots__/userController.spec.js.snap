// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Users Create Should create a user 1`] = `
Object {
  "message": "Usu√°rio criado!",
  "user": Object {
    "__v": 0,
    "_id": "ObjectId",
    "name": "name",
    "username": "TEST",
  },
}
`;

exports[`Users Create Should not create a user with the same username 1`] = `
Object {
  "error": Object {
    "code": 11000,
    "driver": true,
    "errmsg": "E11000 duplicate key error collection: treinamento.users index: username_1 dup key: { : \\"TEST\\" }",
    "index": 0,
    "message": "E11000 duplicate key error collection: treinamento.users index: username_1 dup key: { : \\"TEST\\" }",
    "name": "MongoError",
  },
  "message": "Deu erro!",
}
`;

exports[`Users Create Should not create a without username 1`] = `
Object {
  "error": Object {
    "errors": Array [
      "username is a required field",
    ],
    "inner": Array [],
    "message": "username is a required field",
    "name": "ValidationError",
    "params": Object {
      "path": "username",
    },
    "path": "username",
    "type": "required",
    "value": Object {
      "name": "name",
    },
  },
  "message": "Deu erro!",
}
`;

exports[`Users Find Should find a user 1`] = `
Object {
  "message": "Usuario encontrados",
  "user": Object {
    "__v": 0,
    "_id": "ObjectId",
    "name": "name",
    "username": "TEST2",
  },
}
`;
